### PR TITLE
php 8.3.0 php@8.2 8.2.13 (new formula)

### PR DESCRIPTION
**Pull request by derrabus**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Replaces #155306
Replaces #155312
Obsoletes #155271

The PR includes a fix for PHP CS Fixer which is not yet ready for PHP 8.3. We've solved this issue last year the same way as I did here, see c2009fd541842386e443dcb33db0a5386e47fd31 and e0b0f37e8f8a8db9ed25c44e145a61ac71bd6ef3.

### CODE REVIEW

File: Formula/p/php.rb [hunk 1]
```diff
@@ -2,11 +2,10 @@ class Php < Formula
   desc "General-purpose scripting language"
   homepage "https://www.php.net/"
   # Should only be updated if the new version is announced on the homepage, https://www.php.net/
-  url "https://www.php.net/distributions/php-8.2.12.tar.xz"
-  mirror "https://fossies.org/linux/www/php-8.2.12.tar.xz"
-  sha256 "e1526e400bce9f9f9f774603cfac6b72b5e8f89fa66971ebc3cc4e5964083132"
+  url "https://www.php.net/distributions/php-8.3.0.tar.gz"
+  mirror "https://fossies.org/linux/www/php-8.3.0.tar.xz"
+  sha256 "557ae14650f1d1984d3213e3fcd8d93a5f11418b3f8026d3a2d5022251163951"
```
Comment (RSickenberg) [inline Formula/p/php.rb:7]:
```suggestion
  url "https://www.php.net/distributions/php-8.3.0.tar.xz"
  mirror "https://fossies.org/linux/www/php-8.3.0.tar.xz"
  sha256 "1db84fec57125aa93638b51bb2b15103e12ac196e2f960f0d124275b2687ea54"
```

**Comment by derrabus** [comment]
PR is ready. 🎉 

Thank you for your help, @RSickenberg, @cho-m and @chenrui333!

File: Formula/p/php@8.2.rb [hunk 2]
```diff
@@ -0,0 +1,474 @@
+class PhpAT82 < Formula
+  desc "General-purpose scripting language"
+  homepage "https://www.php.net/"
+  # Should only be updated if the new version is announced on the homepage, https://www.php.net/
+  url "https://www.php.net/distributions/php-8.2.13.tar.xz"
+  mirror "https://fossies.org/linux/www/php-8.2.13.tar.xz"
+  sha256 "2629bba10117bf78912068a230c68a8fd09b7740267bd8ebd3cfce91515d454b"
+  license "PHP-3.01"
+  revision 1
```
Comment (ZhongRuoyu) [inline Formula/p/php@8.2.rb:9]: We don't need the revision here.

```suggestion
```

**Review by ZhongRuoyu — COMMENTED** [review]
Thanks @derrabus!

File: Formula/p/php@8.2.rb (hunk 2 above)
Comment (derrabus) [reply Formula/p/php@8.2.rb]: Right, thanks. 🙈

**Review by ZhongRuoyu — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]
Thanks!

**Pull request merged**